spec/ruby/core/tracepoint/inspect_spec.rb

```diff
@@ -31,6 +78,34 @@ class TracePointSpec::C
       end
     end
 
-    inspect.should == "#<TracePoint:class@#{__FILE__}:#{line}>"
+    inspect.should == "#<TracePoint:class #{__FILE__}:#{line}>"
+  end
+
+  it 'returns a String showing the event and thread for :thread_begin event' do
+    inspect = nil
+    thread_inspection = nil
+    TracePoint.new(:thread_begin) { |tp|
```

Comment:
Can you use a check similar to `next unless TracePointSpec.target_thread?` but for `thread` here?
Otherwise the TracePoint might catch other Threads which is unwanted as it would fail the spec randomly.
I'm thinking
```ruby
thread = nil
TracePoint.new(:thread_begin) { |tp|
  next unless Thread.current == thread
```